### ISSUE TITLE
`is_nan` does not work with `gfortran-8 -O3 -ffast-math -Ofast -flto` on Kunpeng Platform with Ubuntu 18.04 Linux 4.15.0-166-generic

### ISSUE THREAD

**Issue by zaikunzhang** [equipez/infnan#2]
On [Kunpeng Platform](https://e.huawei.com/hk/products/servers/computing-kunpeng) with Ubuntu 18.04 Linux 4.15.0-166-generic, `is_nan` fails to identify `NaN` when compiled with `gfortran-8 -O3 -ffast-math -Ofast -flto` (`ieee_is_nan` does not work either).

The details are as follows. 

```
$ lscpu
Architecture:        aarch64
Byte Order:          Little Endian
CPU(s):              2
On-line CPU(s) list: 0,1
Thread(s) per core:  1
Core(s) per socket:  2
Socket(s):           1
NUMA node(s):        1
Vendor ID:           0x48
Model:               0
Stepping:            0x1
BogoMIPS:            200.00
L1d cache:           64K
L1i cache:           64K
L2 cache:            512K
L3 cache:            32768K
NUMA node0 CPU(s):   0,1
Flags:               fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma dcpop asimddp asimdfhm

$ uname -a 
Linux kunpeng-0001 4.15.0-166-generic #174-Ubuntu SMP Wed Dec 8 19:08:37 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux

$ gfortran --version
GNU Fortran (Ubuntu/Linaro 8.4.0-1ubuntu1~18.04) 8.4.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ cd test && make clean && make gtest
rm -f *.o *.mod *.dbg *test
gfortran -O3 -ffast-math -Ofast -flto -o gtest ../consts.f90 ../infnan.f90 ieee_infnan.f90 testinfnan.f90 test.f90
./gtest

Single-precision tests:
 IS_NAN:  F F F F F F F F F F F F F F F F F F F
 IEEE_IS_NAN:  F F F F F F F F F F F F F F F F F F F
Some tests failed. The data is
Infinity    -Inf     NaN     NaN  4.E+01 -4.E+01  0.E+00 -0.E+00  0.E+00 -0.E+00  3.E+38 -3.E+38  1.E-07 -1.E-07  1.E-38 -1.E-38  0.E+00  1.E+00 -1.E+00

Double-precision tests:
 IS_NAN:  F F F F F F F F F F F F F F F F F F F
 IEEE_IS_NAN:  F F F F F F F F F F F F F F F F F F F
Some tests failed. The data is
Infinity    -Inf     NaN     NaN  4.D+01 -4.D+01  1.D+00 -1.D+00  0.D+00 -0.D+00  2.+308 -2.+308  2.D-16 -2.D-16  2.-308 -2.-308  0.D+00  1.D+00 -1.D+00
 
Note: The following floating-point exceptions are signalling: IEEE_INVALID_FLAG IEEE_OVERFLOW_FLAG
STOP 1
Makefile:65: recipe for target 'gtest' failed
make: *** [gtest] Error 1
```

**Comment by zaikunzhang** [comment]
Is this also resolved by [024fd17](https://github.com/equipez/infnan/commit/024fd172302b275398cfe6af76677e7f3147191e), the same as iussuse #2 and #3? To be tested when a Kunpeng machine is available.

**Comment by zaikunzhang** [comment]
Closed since gfortran 8 is too old.
